File: fluentcorej9/src/main/java/com/mcreater/fxfluent/util/ReflectUtil.java
package com.mcreater.fxfluent.util;

import java.lang.reflect.Method;

public class ReflectUtil {
    public static boolean openModuleToAllUnnamed(Module targetModule, String... packageNames) {
        try {
            Method method = Module.class.getDeclaredMethod("implAddOpensToAllUnnamed", String.class);
            method.setAccessible(true);
            for (String p : packageNames) method.invoke(targetModule, p);
            return true;
        }
        catch (Exception e) {
            e.printStackTrace();
            System.out.println("Open module failed, please check your command args and add these args: \n--add-opens\njava.base/java.lang=ALL-UNNAMED");
        }
        return false;
    }

    public static boolean exportModuleToAllUnnamed(Module targetModule, String... packageNames) {
        try {
            Method method = Module.class.getDeclaredMethod("implAddExportsToAllUnnamed", String.class);
            method.setAccessible(true);
            for (String p : packageNames) method.invoke(targetModule, p);
            return true;
        }
        catch (Exception e) {
            e.printStackTrace();
            System.out.println("Open module failed, please check your command args and add these args: \n--add-opens\njava.base/java.lang=ALL-UNNAMED");
        }
        return false;
    }
}

File: fluentcorej9/src/main/java/com/mcreater/fxfluent/util/ReflectUtil.kt
package com.mcreater.fxfluent.util

class ReflectUtil {
    companion object {
        @JvmStatic
        fun openModuleToAllUnnamed(targetModule: Module?, vararg packageNames: String?): Boolean {
            try {
                val method =
                    Module::class.java.getDeclaredMethod("implAddOpensToAllUnnamed", String::class.java)
                method.setAccessible(true)
                for (p in packageNames) method.invoke(targetModule, p)
                return true
            } catch (e: Exception) {
                e.printStackTrace()
                println("Open module failed, please check your command args and add these args: \n--add-opens\njava.base/java.lang=ALL-UNNAMED")
            }
            return false
        }

        @JvmStatic
        fun exportModuleToAllUnnamed(targetModule: Module?, vararg packageNames: String?): Boolean {
            try {
                val method =
                    Module::class.java.getDeclaredMethod("implAddExportsToAllUnnamed", String::class.java)
                method.setAccessible(true)
                for (p in packageNames) method.invoke(targetModule, p)
                return true
            } catch (e: Exception) {
                e.printStackTrace()
                println("Open module failed, please check your command args and add these args: \n--add-opens\njava.base/java.lang=ALL-UNNAMED")
            }
            return false
        }
    }
}